auth/src/main/java/com/firebase/ui/auth/ui/email/CheckEmailFragment.java

```diff
@@ -94,6 +97,26 @@ public static CheckEmailFragment getInstance(@NonNull FlowParameters flowParamet
         return fragment;
     }
 
+    public static Task<String> fetchTopProvider(FirebaseAuth auth, @NonNull String email) {
+        if (TextUtils.isEmpty(email)) {
+            return Tasks.forException(new IllegalArgumentException("Email cannot be empty"));
+        }
+
+        return auth.fetchProvidersForEmail(email)
+                .addOnFailureListener(
+                        new TaskFailureLogger(TAG, "Error fetching providers for email"))
+                .continueWith(new Continuation<ProviderQueryResult, String>() {
+                    @Override
+                    public String then(@NonNull Task<ProviderQueryResult> task) throws Exception {
+                        if (!task.isSuccessful()) return null;
+
+                        List<String> providers = task.getResult().getProviders();
+                        if (providers == null || providers.isEmpty()) return null;
```

Comment:
@samtstern Lol, of course not. 😆 I started working on the refactor and I'll post an explanatory comment once the method is in stable place where my comment won't be folded away.

And yeah, I'll add braces. 😄